### PR TITLE
Implement most icons from https://github.com/Peltoche/lsd/issues/764

### PR DESCRIPTION
**Pull request by TruncatedDinosour**
- PKGBUILD -- Arch Linux package manifests

Didn't implement these because I couldn't find a good icon or is just downright impossible to implement

- config.\* -- Configuration files
- Makefile\* -- Make config
- .\*_history -- History files for readline-like - .gdb_history -- GDB debugger history - .lesshst -- `less` history
- Kconfig.\* -- Specific kernel config file
- Kbuild.\* -- Specific kernel build file
- Makefile\* -- Make config
- TODO\* -- TODOs - FAQ\* -- FAQ - LEGACY\* -- Legacy statement
- var/ -- Variable files
- mnt/ -- Mounts
- proc/ -- Processes
- usr/ -- Universal system repository
- dev/ -- Device files
- boot/ -- Bootloader files
- opt/ -- Optional applications
- sys/ -- System files
- src/ -- source code
- tests/ -- Tests for code - test/ -- Test dir
- Media/ -- Media dir
- rc.conf -- OpenRC config - init.d/ -- OpenRC services
- content/ -- Content dir
- page/ -- Pages
- resources/ -- Resources for programs - static/ -- Static resources
- completions/ -- Shell completions
- target/ -- Target files
- dist/ -- Destination files - dest/ -- Destination files
- .local/ -- Local files - local/ -- Local files

Signed-off-by: Ari Archer <ari.web.xyz@gmail.com>

---
#### TODO

- [ ] Use `cargo fmt`
- [ ] Add necessary tests
- [ ] Add changelog entry
- [ ] Update default config/theme in README (if applicable)
- [ ] Update man page at lsd/doc/lsd.md (if applicable)

### CODE REVIEW

File: src/theme/icon.rs [hunk 1]
```diff
@@ -302,55 +386,66 @@ impl IconTheme {
...
             ("json", "\u{e60b}"),            // ""
+            ("js", "\u{e74e}"),              // ""
             ("jsx", "\u{e7ba}"),             // ""
-            ("jl", "\u{e624}"),              // ""
             ("key", "\u{e60a}"),             // ""
             ("kt", "\u{e634}"),              // ""
             ("kts", "\u{e634}"),             // ""
             ("ksh", "\u{f489}"),             // ""
-            ("ld", "\u{e624}"),              // ""
+            ("kts", "\u{e634}"),             // ""
```
Comment (meain) [inline src/theme/icon.rs:423]: Looks like this is repeated.

File: src/theme/icon.rs [hunk 2]
```diff
@@ -62,7 +62,7 @@ impl ByType {
             file: "\u{1f4c4}".into(),
             pipe: "\u{1f4e9}".into(),
             socket: "\u{1f4ec}".into(),
-            executable: "\u{1f3d7}".into(),
```
Comment (meain) [inline src/theme/icon.rs:65]: Let's stick to the existing one for this.

File: src/theme/icon.rs (hunk 1 above)
Comment (TruncatedDinosour) [reply src/theme/icon.rs:423]: > Looks like kt and kts is repeated.

won't it cause another merge conflict if i remove it

File: src/theme/icon.rs (hunk 2 above)
Comment (TruncatedDinosour) [reply src/theme/icon.rs:65]: > Let's stick to the existing one for this.

weird, it was normal before

File: src/theme/icon.rs (hunk 1 above)
Comment (meain) [reply src/theme/icon.rs:423]: No, it should be fine. The issues is mostly when you merge main into your branch. But don't bother about it now, we've enabled suash merge and I can do that from my end.

**Comment by meain** [comment]
I've created https://github.com/Peltoche/lsd/pull/779 with fixes on top of your changes. Closing this for now. Thanks for the PR :D

**Pull request closed without merging**